hack/verify-golint.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+
+array_contains () {
+    local seeking=$1; shift
+    local in=1
+    for element; do
+        if [[ "$element" == "$seeking" ]]; then
+            in=0
+            break
+        fi
+    done
+    return $in
+}
+
+export IFS=$'\n'
+all_packages=(
+	$(go list -f '{{.ImportPath}}' ./... | grep -v vendor | sed 's/github.com\/kubernetes\/kubernetes\///g' | sed 's/k8s.io\/kubernetes\///g')
+)
+linted_file="${KUBE_ROOT}/hack/.linted_packages"
+linted_packages=(
+	$(cat $linted_file)
+)
+unset IFS
+linted=()
+errors=()
+for p in "${all_packages[@]}"; do
+	# Run golint on package/*.go file explicitly to validate all go files
+	# and not just the ones for the current platform.
+	failedLint=$(golint "$p"/*.go)
+	if [ "$failedLint" ]; then
+		if array_contains "$p" "${linted_packages[@]}"; then
+			errors+=( "$failedLint" )
+		fi
+	else
+		array_contains "$p" "${linted_packages[@]}" || linted+=( "$p" )
+	fi
+done
+
+# Check to be sure all the packages that should pass lint are.
+if [ ${#errors[@]} -eq 0 ]; then
+	echo 'Congratulations!  All Go source files have been linted.'
+else
+	{
+		echo "Errors from golint:"
+		for err in "${errors[@]}"; do
+			echo "$err"
+		done
+		echo
+		echo 'Please fix the above errors. You can test via "golint" and commit the result.'
+		echo
+	} >&2
+	false
+fi
+
+# check to make sure all packages that pass lint are in the linted file.
```

Comment:
:+1: Interesting concept. I can only hope that failing because you've made something more awesome doesn't trip anybody up. Should at least be strictly more pleasant than finding another flake.